core/flyout.js

```diff
@@ -510,6 +510,14 @@ Blockly.Flyout.prototype.reflow = function() {
         var blockXY = block.getRelativeToSurfaceXY();
         block.flyoutRect_.setAttribute('x',
             this.RTL ? blockXY.x - blockHW.width + tab : blockXY.x - tab);
+        // For hat blocks we want to shift them down by the hat height
+        // since the y coordinate is the corner, not the top of the
+        // hat.
+        var hatOffset =
+          block.startHat_ ? Blockly.BlockSvg.START_HAT_HEIGHT : 0;
+        if (hatOffset != 0) {
```

Comment:
In JS you can just say 'if (hatOffset) {' for a 3 byte win (post compile).
Don't worry about it, I'll fix this post-merge.